Tramitacao customizada ordemdia expediente

joaohortsenado opened the pull request:
opção de seleção de tramitação customizada para matérias listadas na ordem do dia e expediente, alteração dos relatorios de pauta de sessão para exibir a tramitação customizada ou a ultima tramitação antes da hora da sessão.

## Descrição
<!--- Descreva suas alterações detalhadamente -->

## _Issue_ Relacionada
<!--- Este projeto apenas aceita _pull requests_ relacionadas à _issues_ abertas. -->
<!--- Se está sugerindo uma nova _feature_ ou mudança, por favor discuta em uma _issue_ antes. -->
<!--- Se está corrigindo um _bug_, deve haver uma _issue_ descrevendo-o com passos para reproduzir. -->
<!--- Por favor, adicione o link para a _issue_ aqui: -->

## Motivação e Contexto
<!--- Por que essa mudança é necessária? Qual problema ela resolve? -->

## Como Isso Foi Testado?
<!--- Por favor, descreva detalhadamente como você testou suas mudanças. -->
<!--- Inclua detalhes do seu ambiente de teste e os testes que você executou -->
<!--- para ver como a sua alteração afeta outras áreas do código, etc. -->

## Capturas de Tela (se apropriado):
<!--- Insira imagens, se apropriado, da adição e/ou correção que foi feita -->

## Tipos de Mudanças
<!--- Quais os tipos de alterações introduzidos pelo seu código? Coloque um `x` em todas as caixas que se aplicam: -->
- [ ] _Bug fix_ (alteração que corrige uma _issue_ e não altera funcionalidades já existentes)
- [x ] Nova _feature_ (alteração que adiciona uma funcionalidade e não altera funcionalidades já existentes)
- [ ] Alteração disruptiva (_Breaking change_) (Correção ou funcionalidade que causa alteração nas funcionalidades existentes)

## Checklist:
<!--- Passe por todos os pontos a seguir e coloque um `x` em todas as caixas que se aplicam. -->
<!--- Se você não tem certeza sobre nenhum destes, não hesite em perguntar. Nós estamos aqui para ajudar! -->
- [ x] Eu li o documento de Contribuição (**CONTRIBUTING**).
- [ x] Meu código segue o estilo de código deste projeto.
- [ x] Minha alteração requer uma alteração na documentação.
- [ ] Eu atualizei a documentação de acordo.
- [ ] Eu adicionei testes para cobrir minhas mudanças.
- [ ] Todos os testes novos e existentes passaram.

Commit:
Campo tramitação para expediente e ordem do dia das pautas de sessoes plenarias

Commit:
Migration para tramitacao customizada de ordem dia e expediente da pauta de sessao

joaohortsenado commented:
@LeandroJatai por favor, poderia revisar este PR? Conforme conversamos há algum tempo, eu fiz a implementação de um campo para tramitação customizada da Pauta de sessões. Especificamente, como eu queria um select com apenas as tramitações existentes para a matéria que está sendo incluída na ordem do dia, acabei criando uma solução de extender o choicefield para evitar os erros da validação. Caso você quiser a gente pode reunir pra te explicar as minhas dúvidas em relação a melhor forma de implementação disso.

LeandroJatai commented on sapl/sessao/views.py:
```diff
@@ -839,6 +841,23 @@ def recuperar_materia(request):
     return response
 
+def recuperar_tramitacao(request):
+    tipo = TipoMateriaLegislativa.objects.get(pk=request.GET['tipo_materia'])
```
```suggestion
    tipo = request.GET['tipo_materia']
```

LeandroJatai commented on sapl/sessao/views.py:
```diff
@@ -839,6 +841,23 @@ def recuperar_materia(request):
+    ano = request.GET['ano_materia']
+
+    try:
+        materia = MateriaLegislativa.objects.get(tipo=tipo,
```
```suggestion
        materia = MateriaLegislativa.objects.get(tipo_id=tipo,
```

LeandroJatai commented:
@joaohortsenado, o código me parece ok... Ainda hj baixarei o branch e analisarei a solução implementada, dai a gente se reúne!

LeandroJatai committed:
ajuste automático feito pela IDE para pep8

LeandroJatai committed:
Ajuste de usabilidade

- encaminha além da descrição do status, também a data de tramitação, unidade local e unidade de destino de cada tramitação
- recebe os dados acima na interface, monta options com data, e status.
- qualquer alteração no select, monta um alert, assim como para matéria selecionada, com os dados cima da tramitação

LeandroJatai commented:
@joaohortsenado ficou bacana, fiz algumas alterações de usabilidade:
commit: https://github.com/interlegis/sapl/pull/3584/commits/ab6d699f25bc252f16f7db24185c274c979a087f
Ajuste de usabilidade
- encaminha além da descrição do status, também a data de tramitação, unidade local e unidade de destino de cada tramitação
- recebe os dados acima na interface, monta options com data, e status.
- qualquer alteração no select, monta um alert, assim como para matéria selecionada, com os dados cima da tramitação
![image](https://user-images.githubusercontent.com/9630623/182270362-85538ede-1683-4ec8-bbfa-59345f0a0185.png)
